mopro-ffi/src/app_config/android.rs

```diff
@@ -74,45 +64,46 @@ pub fn build() {
     cleanup_tmp_local(&build_dir);
 }
 
-fn build_for_arch(arch: &str, build_dir: &Path, bindings_out: &Path, mode: &str) {
-    install_arch(arch.to_string());
+fn build_for_arch(arch: AndroidArch, build_dir: &Path, bindings_out: &Path, mode: Mode) {
+    let arch_str = arch.as_str();
+    install_arch(arch_str.to_string());
 
     let mut build_cmd = Command::new("cargo");
     build_cmd
         .arg("ndk")
         .arg("-t")
-        .arg(arch)
+        .arg(arch_str)
         .arg("build")
         .arg("--lib");
-    if mode == "release" {
+    if mode == Mode::Release {
         build_cmd.arg("--release");
     }
     build_cmd
         .env("CARGO_BUILD_TARGET_DIR", build_dir)
-        .env("CARGO_BUILD_TARGET", arch)
+        .env("CARGO_BUILD_TARGET", arch_str)
         .spawn()
         .expect("Failed to spawn cargo build")
         .wait()
         .expect("cargo build errored");
 
     let folder = match arch {
-        "x86_64-linux-android" => "x86_64",
-        "i686-linux-android" => "x86",
-        "armv7-linux-androideabi" => "armeabi-v7a",
-        "aarch64-linux-android" => "arm64-v8a",
-        _ => panic!("Unknown target architecture: {}", arch),
+        AndroidArch::X8664Linux => ARCH_X86_64,
+        AndroidArch::I686Linux => ARCH_I686,
+        AndroidArch::Armv7LinuxAbi => ARCH_ARM_V7_ABI,
+        AndroidArch::Aarch64Linux => ARCH_ARM_64_V8,
     };
 
     let out_lib_path = build_dir.join(format!(
         "{}/{}/{}/libmopro_bindings.so",
         build_dir.display(),
-        arch,
-        mode
+        arch_str,
+        mode.as_str()
     ));
     let out_lib_dest = bindings_out.join(format!("jniLibs/{}/libuniffi_mopro.so", folder));
 
+    println!("COPY: {:?} -> {:?}", out_lib_path, out_lib_dest);
```

Comment:
Just curious, is it left-over?

Reply:
Nice catch, fixed in [9dd8197](https://github.com/zkmopro/mopro/pull/297/commits/9dd81972de930db585b601cac59aded0ac17f880).